include/libhal-arm-mcu/stm32f1/adc.hpp

```diff
@@ -83,7 +68,8 @@ class adc final : public hal::adc
   virtual ~adc() = default;
 
 private:
-  channel get_predefined_channel_info(std::uint8_t p_channel);
+  pins const m_pin;
```

Comment:
```suggestion
  pins m_pin;
```

No const non-static member variables. I'd rather we stick with this even though we delete the copy & move constructor.